hazelcast-sql/src/main/java/com/hazelcast/sql/impl/calcite/schema/MapTableStatistic.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.calcite.schema;
+
+import org.apache.calcite.rel.RelCollation;
+import org.apache.calcite.rel.RelDistribution;
+import org.apache.calcite.rel.RelDistributionTraitDef;
+import org.apache.calcite.rel.RelReferentialConstraint;
+import org.apache.calcite.schema.Statistic;
+import org.apache.calcite.util.ImmutableBitSet;
+
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Simple table statistics for IMap and ReplicatedMap.
+ */
+public class MapTableStatistic implements Statistic {
+    /** Row count that is fixed for the duration of query optimization process. */
+    private final Long rowCount;
+
+    public MapTableStatistic(long rowCount) {
+        this.rowCount = rowCount;
+    }
+
+    @Override
+    public Double getRowCount() {
+        return (double) rowCount;
```

Comment:
It'd be nice to have an explanation somewhere why rowCount is `double` (not a `long`)?

Reply:
This comes from a Calcite interface, so we do not have any control of it.